pkg/ingester/ingester_v2.go

```diff
@@ -1512,12 +1572,21 @@ func (i *Ingester) shipBlocks(ctx context.Context) {
 		}
 		defer userDB.casState(activeShipping, active)
 
-		if uploaded, err := userDB.shipper.Sync(ctx); err != nil {
+		uploaded, err := userDB.shipper.Sync(ctx)
+		if err != nil {
 			level.Warn(util.Logger).Log("msg", "shipper failed to synchronize TSDB blocks with the storage", "user", userID, "uploaded", uploaded, "err", err)
 		} else {
 			level.Debug(util.Logger).Log("msg", "shipper successfully synchronized TSDB blocks with storage", "user", userID, "uploaded", uploaded)
 		}
 
+		// The shipper meta file could be updated even if the Sync() returned an error,
```

Comment:
Shipper meta file may have changed even if no blocks were uploaded -- when shipper has removed obsolete entries from the list. I guess we don't need to care about that.

Reply:
Yes, right. Shouldn't be a problem for our use case because technically the blocks removed from the meta because of the retention have been shipped anyway. I've updated the code comment.